ports/esp32s2/common-hal/microcontroller/Processor.c

```diff
@@ -34,8 +34,15 @@
 
 #include "soc/efuse_reg.h"
 
+#include "esp-idf/components/driver/esp32s2/include/driver/temp_sensor.h"
+
 float common_hal_mcu_processor_get_temperature(void) {
-    return NAN;
+    float tsens_out;
+    temp_sensor_config_t temp_sensor = TSENS_CONFIG_DEFAULT(); // DEFAULT: range:-10℃ ~  80℃, error < 1℃.
+    temp_sensor_set_config(temp_sensor);
+    temp_sensor_start();
+    temp_sensor_read_celsius(&tsens_out);
```

Comment:
```suggestion
    temp_sensor_read_celsius(&tsens_out);
    temp_sensor_stop();
```

As Jeff mentioned, please add temp_sensor_stop so that this is a self-contained operation.